Add TLS configuration options for auth (#536)
apollographql/apollo-mcp-server: docs/source/config-file.mdx

| `resource_documentation`         | `string`       |         | Optional link to more documentation relating to this MCP server                                    |
| `scopes`                         | `List<string>` |         | List of queryable OAuth scopes from the upstream OAuth servers                                     |
| `disable_auth_token_passthrough` | `bool`         | `false` | Optional flag to disable passing validated Authorization header to downstream API                  |
| `tls.ca_cert`                    | `string`       |         | Path to a CA certificate to trust (PEM format). |

> | `tls.danger_accept_invalid_certs`| `bool`         | `false` | Accept invalid TLS certificates. Warning: Insecure. Use only for development or testing.          |

Reviewer: ```suggestion
| `tls.danger_accept_invalid_certs`| `bool`         | `false` | Accepts invalid TLS certificates. Set this to `true for development or testing purposes only. |
```

(This might need spacing fixes)

Author: Thanks @mabuyo for reviewing the doc updates! I applied your suggestions and had the file reformatted.
